src/exercises/model.py

```diff
@@ -0,0 +1,39 @@
+from collections import namedtuple
+from pathlib import Path
+from typing import List
+
+
+class Exercise(namedtuple("Exercise", ["name"])):
+    pass
+
+
+class Directory(namedtuple("Directory", ["name", "children"])):
+    def list_exercises(self) -> List[Path]:
+        return self.__inner_list_exercises__("")
+
+    def __inner_list_exercises__(self, prefix) -> List[Path]:
+        dir_prefix = f"{self.name}/"
+
+        res = []
+
+        for child in self.children:
+            if isinstance(child, Exercise):
+                res.append(Path(f"{prefix}{dir_prefix}{child.name}.cairo"))
+            elif isinstance(child, Directory):
```

Comment:
Same question on unmatched cases to catch inconsistencies.

Reply:
done